solidity/contracts/bridge/Wallets.sol

```diff
@@ -517,4 +531,66 @@ library Wallets {
 
         self.registry.closeWallet(wallet.ecdsaWalletID);
     }
+
+    // TODO: Documentation
+    function notifyFundsMoved(
+        Data storage self,
+        bytes20 walletPubKeyHash,
+        bytes32 targetWalletsHash
+    ) external {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        // Check that the wallet is in the MovingFunds state but don't check
+        // if the moving funds timeout is exceeded. That should give a
+        // possibility to move funds in case when timeout was hit but was
+        // not reported yet.
+        require(
+            wallet.state == WalletState.MovingFunds,
+            "ECDSA wallet must be in MovingFunds state"
+        );
+
+        uint32 commitmentSubmittedAt = wallet.movingFundsCommitmentSubmittedAt;
+        require(
+            commitmentSubmittedAt > 0,
+            "Target wallet commitment not submitted yet"
+        );
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp >
+                commitmentSubmittedAt +
+                    self.movingFundsCommitmentChallengePeriod,
+            "Target wallet commitment challenge period has not passed yet"
+        );
+
+        // Make sure that the target wallets where funds were moved to are
+        // exactly the same as the ones the source wallet committed to.
+        require(
+            wallet.movingFundsTargetWalletsCommitmentHash == targetWalletsHash,
+            "Target wallets don't correspond to the commitment"
+        );
+
+        // If funds were moved, the wallet has no longer a main UTXO.
+        delete wallet.mainUtxoHash;
+
+        closeWallet(self, walletPubKeyHash);
+    }
+
+    // TODO: Function for committing target wallets for moving funds.
+    //       Can be called only when `movingFundsRequestedAt` field is
+    //       grater than zero, `movingFundsCommitmentSubmittedAt` and
+    //       `movingFundsTargetWalletsCommitmentHash` fields are zeroed, and
+    //       the caller is a wallet's operator. If validation passes, it must
+    //       set the `movingFundsCommitmentSubmittedAt` and
+    //       `movingFundsTargetWalletsCommitmentHash` fields accordingly.
```

Comment:
Shouldn't we also check if target wallets are LIVE?

Reply:
Not `Live` but `MovingFunds`. However, this is implied by the wallet lifecycle. If `movingFundsRequestedAt` is `> 0` that means something caused the `moveFunds` internal function to be called. That function can be called only on `Live` wallets and makes the state transition to be `MovingFunds`. That said, there is no point to check the wallet state explicitly within the function specified by this TODO as this is redundant in that case.

Reply:
Yeah, I was talking about the source wallet and misunderstood you. Regarding target wallets, you're right. However, the purpose of this TODO is just to capture low-level technical details related to this PR and not a full specification of those functions since we have that in the RFC.